### ISSUE TITLE
TextInput does not allow mutating value after initialization

### ISSUE THREAD

**Issue by ethndotsh** [tremorlabs/tremor-npm#251]
I have a button that my users can press to get suggestions on what to input (basically generates random data then populates a text input). I am trying to use `TextInput` but am finding that it is not working how I expected it to.

If I use `useState` to set a variable, e.g. `myInputValue`, that I mutate when they press my button, and set the `value` prop of the `TextInput` to `myInputValue`, I expect the `TextInput`'s value to change when `myInputValue` is mutated. It currently does not.

A way to fix this would be to add a `useEffect` that tracks when the `value` prop changes, and change the internal `inputValue` state variable when event is fired. E.g:

```js
useEffect(() => {
  setInputValue(value)
}, [value])
```

I am not sure if this is intentional or not, but I just wanted to make sure that this curious behaviour was known.

**Comment by mitrotasios** [comment]
Hey @cursecodes, thanks for raising this issue. We will look into this soon. In the meantime maybe try adding a `key` prop to the component with your input value as the key prop's value. This should re-render the component every time the value changes. Sorry about that!

```
const [inputValue, setInputValue] = useState("");

return <TextInput key={ inputValue } value={ inputValue } ... />
```

**Comment by mitrotasios** [comment]
fixed in `v1.3.1`, thanks again!
